config-bootstrapper/src/main/java/org/hypertrace/core/bootstrapper/ConfigBootstrapper.java

```diff
@@ -29,9 +32,15 @@ public static void main(String[] args) {
    * */
   public static BootstrapRunner bootstrapper(BootstrapArgs bootstrapArgs) {
     Config config = ConfigFactory.parseFile(new File(bootstrapArgs.getConfigFile())).resolve();
-    String dataStoreType = config.getString(DocumentStoreConfig.DATASTORE_TYPE_CONFIG_KEY);
+    Config documentStoreConfig = config.getConfig("document.store");
+    DatastoreConfig datastoreConfig = TypesafeConfigDatastoreConfigExtractor.from(documentStoreConfig, DATASTORE_TYPE_CONFIG_KEY)
+            .extract();
     Datastore datastore =
-        DatastoreProvider.getDatastore(dataStoreType, config.getConfig(dataStoreType));
+        DatastoreProvider.getDatastore(datastoreConfig);
+
+//    grpcServiceContainerEnvironment.getLifecycle().shutdownComplete().thenRun(datastore::close);
+//    This seems like a shutdown hook, can it go in finalizeBootstrapper()? @suresh
```

Comment:
When does "finalizeBootstrapper()" happen?

Reply:
It is added as a shutdownHood

`Runtime.getRuntime().addShutdownHook(new Thread(ConfigBootstrapper::finalizeBootstrapper));`

On line 48